src/main/java/org/gridsuite/study/server/service/NetworkConversionService.java

```diff
@@ -92,15 +92,23 @@ public String getExportFormats() {
         return restTemplate.exchange(networkConversionServerBaseUri + path, HttpMethod.GET, null, typeRef).getBody();
     }
 
-    public ExportNetworkInfos exportNetwork(UUID networkUuid, String variantId, String nodeName, String studyName, String format, String paramatersJson) {
+    /**
+     * @param fileName name of the exported file. If null, a file name is generated from the study and node names
+     */
+    public ExportNetworkInfos exportNetwork(UUID networkUuid, String variantId, String nodeName, String studyName,
+                                            String format, String paramatersJson, String fileName) {
         var uriComponentsBuilder = UriComponentsBuilder.fromPath(DELIMITER + NETWORK_CONVERSION_API_VERSION
                 + "/networks/{networkUuid}/export/{format}");
         if (!variantId.isEmpty()) {
             uriComponentsBuilder.queryParam("variantId", variantId);
         }
 
-        if (!StringUtils.isEmpty(studyName) && !StringUtils.isEmpty(nodeName)) {
-            uriComponentsBuilder.queryParam("fileName", URLEncoder.encode(studyName + "_" + nodeName, StandardCharsets.UTF_8));
+        String finalFilename = fileName;
+        if (StringUtils.isEmpty(finalFilename) && !StringUtils.isEmpty(studyName) && !StringUtils.isEmpty(nodeName)) {
+            finalFilename = URLEncoder.encode(studyName + "_" + nodeName, StandardCharsets.UTF_8);
+        }
```

Comment:
In my opinion, studyName and nodeName now have no role, a fileName is enough. Too many optional params for the same objective make the code difficult to maintain.

Reply:
I think we need to construct it in backed for case of zip file exported, to be tested

Reply:
I didn't find how to export a zip file (and use this function).